02_data-wrangling.Rmd

```diff
@@ -0,0 +1,191 @@
+author: "Katie Willi"
+date: "`r Sys.Date()`"
+output: html_document
+---
+
+```{r setup, include = FALSE}
+knitr::opts_chunk$set(echo=TRUE, eval=TRUE, message=FALSE, warning=FALSE, rows.print=5, fig.width=11)
+```
+
+### Lesson Objectives
+
+In the last lesson, we learned how to pull data from an API and reduce redundancies in our workflows through functions and iteration. In this lesson we will use the functions in the previous lesson to learn how to manipulate data frames with the {tidyverse}, and plot elegant time series graphs with the {ggplot2}, {scales} and {plotly} packages.
+
+There are **five exercises** in this lesson that must be completed.
+
+## Pulling in necessary packages and data sets
+
+```{r}
+library(tidyverse) # ggplot2 is included in the {tidyverse}
+library(httr)
+library(jsonlite)
+library(plotly)
+library(scales)
+```
+
+Using the `parkwide_visitation()` function from the last lesson and mapping, let's pull park-wide visitor data from 1980-2021, and name the final object `parkwide`. (Code hack: we can use `1980:2021` to create a vector of years so we don't have to write each year out!)
+
+```{r}
+parkwide_visitation <- function(year){
+
+# pull in the data
+raw_data <- httr::GET(url = 
+                        # parse out year so that it can be chosen with the "year" argument, using paste0()
+                        paste0("https://irmaservices.nps.gov/v3/rest/stats/total/", year))
+
+# convert content to text
+extracted_data <- httr::content(raw_data, as = "text", encoding = "UTF-8") 
+
+# parse text from JSON to data frame
+final_data <- jsonlite::fromJSON(extracted_data)
+
+return(final_data)
+
+}
+
+years <- (1980:2021)
+
+parkwide <- years %>% 
+  map(~ parkwide_visitation(year = .)) %>% 
+  bind_rows()
+```
+
+### Exercise #1 {style="color: maroon"}
+
+**Using the `unit_visitation()` function from the last lesson and mapping, pull visitor data from 1980-2021 for the following park units: ROMO, ACAD, LAKE, YELL, GRCA, ZION, OLYM, and GRSM. Name the final output `units`.**
+
+```{r}
+
+```
+
+## Exploring our data
+
+Look at the data frame structure of `parkwide` and `units`; they're exactly the same! So let's go ahead and bind those together:
+
+```{r}
+visitation <- bind_rows(parkwide, units)
+```
+
+... except, the rows in `parkwide`'s UnitCode and UnitCode columns are empty. 😑 Let's fix the `UnitCode` column to list "Parkwide" using `mutate()` and an `ifelse()` statement:
+
+```{r}
+visitation <- visitation %>% mutate(UnitCode = ifelse(is.na(UnitCode), "Parkwide", UnitCode))
+```
+
+Think of the above `ifelse()` operation as: "If the column `UnitCode` is `NA`, replace `NA` with `Parkwide`. Otherwise, preserve what is already in the `UnitCode` column."
+
+Now that we have a single data set containing all of the NPS visitation data that we've pulled, let's start exploring it! But first, let's aggregate the monthly data into annual data using `group_by()` and `summarize()`:
+
+```{r}
+yearly <- visitation %>%
```

Comment:
This might be an annoying fix since you use it so much downstream, but thinking since above they create a variable 'years', having this one as 'yearly' will likely get confusing. Maybe change to 'annual_visitation' or something more self-explanatory.